app/features/wallet/wallet.tsx

```diff
@@ -12,15 +13,52 @@ import { useTheme } from '../theme';
 import { type AuthUser, useHandleSessionExpiry } from '../user/auth';
 import { useUpsertUser, useUser } from '../user/user-hooks';
 import { TaskProcessor, useTakeTaskProcessingLead } from './task-processing';
+// import { agicashDb } from '../agicash-db/database';
+
+type SupabaseInitializationState =
+  | {
+      status: 'initializing' | 'initialized';
+    }
+  | {
+      status: 'error';
+      error: Error;
+    };
 
 const useInitializeSupabaseSessionStore = () => {
   const { generateThirdPartyToken } = useOpenSecret();
+  const [state, setState] = useState<SupabaseInitializationState>({
+    status: 'initialized',
+  });
 
   useEffect(() => {
     supabaseSessionStore
       .getState()
       .setJwtGetter(() => generateThirdPartyToken().then((res) => res.token));
+
+    // Needed for this workaround https://github.com/supabase/realtime/issues/282#issuecomment-2630983759
```

Comment:
I don't know why but without making this `realtime.setAuth()` call supabase would end up sending subcribed to postgres message twice and that would break something sometimes. seems to be working fine now